File: src/telegram/types/inline-query.ts
import * as type from '../types'
import Type from './type'


export declare type JsonInlineQuery = {
	id: type.String
	from: type.JsonUser
	query: type.String
	offset: type.Number
	chat_type?: 'sender' | type.ChatType
	location?: type.JsonLocation
}


export default class InlineQuery extends Type {
	id: type.String
	from: type.User
	query: type.String
	offset: type.Number
	chat_type?: 'sender' | type.ChatType
	location?: type.Location

	defaultObject!: JsonInlineQuery

	constructor(inlineQuery: JsonInlineQuery) {
		super()

		this.defaultObject = inlineQuery

		this.id = inlineQuery.id
		this.from = new type.User(inlineQuery.from)
		this.query = inlineQuery.query
		this.offset = inlineQuery.offset
		this.chat_type = inlineQuery.chat_type
		this.location = inlineQuery.location ? new type.Location(inlineQuery.location) : undefined
	}
}

File: telegram-api-json/pygen/InlineQuery.py

from telelib.bot import DefaultType

# @url https://core.telegram.org/bots/api#inlinequery# ['This object represents an incoming inline query. When the user sends an empty query, your bot could return some default or trending results.']
class InlineQuery(DefaultType):

    
	# @var id Unique identifier for this query
	@property
	def id(self) -> String:
		return self._d["id"]
	# @var from Sender
	@property
	def from(self) -> User:
		return self._d["from"]
	# @var query Text of the query (up to 256 characters)
	@property
	def query(self) -> String:
		return self._d["query"]
	# @var offset Offset of the results to be returned, can be controlled by the bot
	@property
	def offset(self) -> String:
		return self._d["offset"]
	# @var chat_type Optional. Type of the chat from which the inline query was sent. Can be either "sender" for a private chat with the inline query sender, "private", "group", "supergroup", or "channel". The chat type should be always known for requests sent from official clients and most third-party clients, unless the request was sent from a secret chat
	@property
	def chat_type(self) -> String:
		return self._d["chat_type"]
	# @var location Optional. Sender location, only for bots that request user location
	@property
	def location(self) -> Location:
		return self._d["location"]

